### DISCUSSION TITLE
tumor - normal data: how to uncouple files?

### DISCUSSION THREAD

**Discussion by complexgenome** [nextflow-io/nextflow#4091] [Q&A]
Hi, 
I'm interested to process tumor-normal pairs. I'd like to process them through different steps: alignment, duplicates, etc. and eventually use pairs for variant calling from different tools.

```
workflow {
        tumorSamples = Channel.fromFilePairs('/sc/arion/projects/sanjeev/test_input_data/MM-*-tumor-*.fastq.gz')
        normalSamples = Channel.fromFilePairs('/sc/arion/projects/sanjeev/test_input_data/MM-*-normal-*.fastq.gz')

tumorSamples
        .combine(normalSamples)
        .map { tumorSample, normalSample -> tuple(path(tumorSample), path(normalSample)) }
        .set { pairedSamples }
align(pairedSamples).view() 
}

process align {
    
        input:
  tuple val(sampleid),path (tumorSamples)
 
        output:
        stdout

        script:
"""

echo ${sampleid}

"""
}

```

I get no output:

> N E X T F L O W  ~  version 22.10.7
> Launching `pairs.nf` [astonishing_franklin] DSL2 - revision: dd7f3c9bc5
> [-        ] process > align -

My files are as follow:

`ls /sc/arion/projects/sanjeev/test_input_data/MM-*-normal-*.fastq.gz
`
> /sc/arion/projects/sanjeev/test_input_data/MM-1392-DNA-normal-01-01_L003_R1_001.fastq.gz
> /sc/arion/projects/sanjeev/test_input_data/MM-1392-DNA-normal-01-01_L003_R2_001.fastq.gz
> /sc/arion/projects/sanjeev/test_input_data/MM-6189-DNA-normal-01-01_L003_R1_001.fastq.gz
> /sc/arion/projects/sanjeev/test_input_data/MM-6189-DNA-normal-01-01_L003_R2_001.fastq.gz

Similarly for tumor files.

What do I do?
Also, would sampleid capture name as `MM-1392-DNA`?

**Comment by bentsherman** [comment]
Inspect the output of `Channel.fromFilePairs()` with the `view` operator to make sure it matches what you expect. It should return a tuple of the form `[glob, files]`. I *think* if you have multiple globs in the file path, it will use the first glob as the sample id.

**Comment by complexgenome** [comment]
@bentsherman 
Thank you for your kind reply.

I was able to find the problem - I'd to provide R1 and R2. I could see contents using view() for normal and tumor.

How do I make/merge tumor-normal files into one pair/couple to send into align?
I try to use combine.map but fail with error of `closure_9_type`.

```

workflow {
        tumorSamples = Channel.fromFilePairs('/sc/arion/projects/pareks02a/sanjeev/test_input_data/MM-*-tumor-*_{R1,R2}*.fastq.gz')
        normalSamples = Channel.fromFilePairs('/sc/arion/projects/pareks02a/sanjeev/test_input_data/MM-*-normal-*_{R1,R2}*.fastq.gz')
normalSamples.view()
tumorSamples.view()
 
tumorSamples.combine(normalSamples).map { tumorSample, normalSample -> tuple(path(tumorSample), path(normalSample)) }.set { pairedSamples }

pairedSamples.view()

}

```
It prints normal and tumor.

> 
> 
> [MM-6189-DNA-tumor-01-01_L003, [/path/MM-6189-DNA-tumor-01-01_L003_R1_001.fastq.gz, /path/MM-6189-DNA-tumor-01-01_L003_R2_001.fastq.gz]]
> [MM-6189-DNA-normal-01-01_L003, [/path/MM-6189-DNA-normal-01-01_L003_R1_001.fastq.gz, /path/MM-6189-DNA-normal-01-01_L003_R2_001.fastq.gz]]
> [MM-1392-DNA-tumor-01-01_L003, [/path/MM-1392-DNA-tumor-01-01_L003_R1_001.fastq.gz, /path/MM-1392-DNA-tumor-01-01_L003_R2_001.fastq.gz]]
> [MM-1392-DNA-normal-01-01_L003, [/path/MM-1392-DNA-normal-01-01_L003_R1_001.fastq.gz, /path/MM-1392-DNA-normal-01-01_L003_R2_001.fastq.gz]]
> Invalid method invocation `call` with arguments: [MM-6189-DNA-tumor-01-01_L003, [/path/MM-6189-DNA-tumor-01-01_L003_R1_001.fastq.gz, /path/MM-6189-DNA-tumor-01-01_L003_R2_001.fastq.gz], MM-6189-DNA-normal-01-01_L003, [/path/MM-6189-DNA-normal-01-01_L003_R1_001.fastq.gz, /path/MM-6189-DNA-normal-01-01_L003_R2_001.fastq.gz]] (java.util.LinkedList) on _closure9 type
